### PR TITLE
Add Magnitude class

### PR DESCRIPTION
**Pull request by cquiroz**
This is a Magnitude class very much based on the one from ocs.

A question for @tpolecat and @swalker2m. I wanted to add right away a list of `Magnitude` to `Target` but it requires me to develop DAOs and importers from ocs2. Are we still planning to use those?

### CODE REVIEW

File: modules/model/shared/src/main/scala/gem/Magnitude.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+/**
+ * Describes the magnitude of a target on a given band
+ */
+@Lenses
+case class Magnitude(
+  value: MagnitudeValue,
+  band: MagnitudeBand,
+  error: Option[MagnitudeValue],
+  system: MagnitudeSystem
```
Comment (swalker2m) [inline modules/model/shared/src/main/scala/gem/Magnitude.scala:21]: I guess we should make this a `final case class`.

File: modules/model/shared/src/main/scala/gem/Magnitude.scala [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  ): Int =
+    (x,y) match {
+      case (Some(m1), Some(m2)) => c.compare(m1, m2)
+      case (None,     None)     =>  0
+      case (_,        None)     => -1
+      case (None,     _)        =>  1
+    }
+
+  // comparison on Option[Magnitude] that reverses the way that None is treated, i.e. None is always > Some(Magnitude).
+  implicit val MagnitudeOptionOrdering: Order[Option[Magnitude]] =
```
Comment (swalker2m) [inline modules/model/shared/src/main/scala/gem/Magnitude.scala:54]: I know this comes from the existing codebase but it strikes me as odd.  I feel like `Option[Magnitude]` is not naturally comparable.

File: modules/model/shared/src/main/scala/gem/Magnitude.scala [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    new Order[Option[Magnitude]] {
+      override def compare(x: Option[Magnitude], y: Option[Magnitude]): Int =
+        compareOptionWith(x, y, MagnitudeOrdering)
+    }
+
+  /** @group Typeclass Instances */
+  implicit val EqMagnitude: Eq[Magnitude] = Eq.by(x => (x.value, x.band, x.error, x.system))
```
Comment (swalker2m) [inline modules/model/shared/src/main/scala/gem/Magnitude.scala:61]: We have `Order[Magnitude]` so we don't need `Eq[Magnitude]`?

Comment (cquiroz) [reply modules/model/shared/src/main/scala/gem/Magnitude.scala:61]: Good catch

File: modules/model/shared/src/main/scala/gem/Magnitude.scala (hunk 2 above)
Comment (cquiroz) [reply modules/model/shared/src/main/scala/gem/Magnitude.scala:54]: We used that for how we sort magnitudes on the UI. It is indeed a bit odd and I was unsure to keep it unless we really needed it. I can remove it

Comment (tpolecat) [reply modules/model/shared/src/main/scala/gem/Magnitude.scala:54]: Yeah this is a UI-related thing that doesn't belong in the model.

File: modules/model/shared/src/main/scala/gem/Magnitude.scala [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  /** group Typeclass Instances */
+  implicit val MagnitudeShow: Show[Magnitude] =
+    Show.show { mag =>
+      val errStr = mag.error.map { e => f" e${e.toDoubleValue}%.2f " }.getOrElse(" ")
+      f"${mag.band.shortName}${mag.value.toDoubleValue}%.2f$errStr(${mag.system.tag})"
+    }
+}
```
Comment (tpolecat) [inline modules/model/shared/src/main/scala/gem/Magnitude.scala:69]: Can we make this the `toString` and have `Show` delegate? So far we have tried to keep `Show` and `toString` consistent, and explicitly never visible to users. Typically we add the constructor around it, like `Magnitude(...)`.

**Pull request merged**